src/taskFlow.ts

```diff
@@ -1,38 +1,40 @@
 import TaskRegistry from "./taskRegistry";
 import * as yaml from "js-yaml";
-import * as fs from 'fs'
+import * as fs from "fs";
 import * as path from "path";
 
 // executor.js
 
 class TaskFlow {
   static taskRegistry: TaskRegistry;
 
- 
-  static configure(ymlPath:string) {
-
+  static configure(ymlPath: string) {
     if (!this.taskRegistry) {
       this.taskRegistry = new TaskRegistry();
     }
 
     this.addTasksFromConfig(ymlPath);
-
   }
 
-
-  static execute(inputData: any) {
+  static async execute(inputData: any) {
     const matchedTask = this.taskRegistry.getMatchingTask(inputData);
 
     if (matchedTask) {
-       // console.log('Executing task:', matchedTask);
-      return matchedTask(inputData);
+      try {
+        // Use `await` to execute the task function
+        const result = await matchedTask(inputData);
+        // Handle the result if needed
+        return result;
+      } catch (error) {
```

Comment:
lets not add the try/catch here. if we add a catch, user will not get exact error that happened in task. 
If you want it for log purpose then log it then throw again. 
also instead of adding try catch. use await func(). catch